### PR TITLE
Delete libSyntax

### PR DESCRIPTION
**Pull request by CodaFi**
Now that swift-syntax no longer consumes the serialized parse tree (see apple/swift-syntax#1041), and we are no longer building the internal swift-syntax dylib (see apple/swift#61898), we are finally able to get rid of the C++ libSyntax library.

Delete libSyntax, the tests dependent upon it, and redistribute some of the remaining useful types it defined to other parts of the compiler.

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
@swift-ci test source compatibility

**Comment by compnerd** [comment]
@swift-ci please build toolchain windows platform

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
@swift-ci please build toolchain windows platform

**Comment by CodaFi** [comment]
@swift-ci test source compatibility

**Comment by CodaFi** [comment]
apple/llvm-project#5630
@swift-ci please build toolchain windows platform

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by AnthonyLatsis** [comment]
Is this inadvertently conflicting with #62132?

**Comment by CodaFi** [comment]
apple/swift-syntax#1084
@swift-ci test macOS

**Comment by CodaFi** [comment]
@AnthonyLatsis @rintaro I suspect I’m going to win this race 😉.

**Comment by AnthonyLatsis** [comment]
>  I suspect I’m going to win this race 

😅
___
`-emit-syntax` is a hidden frontend mode, so I guess we’re not tied by any promises here. Still, how do we go about this [bug report](https://github.com/apple/swift/issues/61880) now that the mode is being removed? Are there any migration directions?

**Comment by compnerd** [comment]
https://github.com/apple/swift-corelibs-foundation/pull/4660 might actually help ... seems that a Foundation change was not tested on Windows.

**Comment by CodaFi** [comment]
> Are there any migration directions?

-emit-syntax was, to my mind, a debug flag for rendering the AST as JSON. I worry that bug reports like that imply that folks were parsing that output instead of using SwiftSyntax itself (wouldn't be the first time - folks have asked us to even stabilize the -dump-ast output). The migration path is to use SwiftSyntax.

**Comment by CodaFi** [comment]
apple/llvm-project#5630
apple/swift-corelibs-foundation#4661
@swift-ci please build toolchain windows platform

File: include/swift/Parse/Token.h [hunk 1]
```diff
@@ -19,13 +19,23 @@
 
 #include "swift/Basic/SourceLoc.h"
 #include "swift/Basic/LLVM.h"
-#include "swift/Syntax/TokenKinds.h"
+#include "swift/Parse/Token.h"
```
Comment (AnthonyLatsis) [inline include/swift/Parse/Token.h:22]: @CodaFi Found a self-include. I wonder why it works though.

**Pull request merged**